src/approaches/delivery_approach.py

```diff
@@ -0,0 +1,100 @@
+"""An approach that implements a delivery-specific policy.
+
+Example command line:
+    python src/main.py --approach delivery_policy --seed 0 \
+        --env pddl_easy_delivery_procedural_tasks \
+        --strips_learner oracle --num_train_tasks 1
+"""
+
+from typing import Callable, cast
+
+from predicators.src import utils
+from predicators.src.approaches import ApproachFailure
+from predicators.src.approaches.nsrt_learning_approach import \
+    NSRTLearningApproach
+from predicators.src.envs.pddl_env import _PDDLEnvState
+from predicators.src.structs import Action, LDLRule, LiftedAtom, \
+    LiftedDecisionList, State, Task
+
+
+class DeliverySpecificApproach(NSRTLearningApproach):
+    """Implements a delivery-specific policy."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "delivery_policy"
+
+    def _solve(self, task: Task, timeout: int) -> Callable[[State], Action]:
+        goal = task.goal
+
+        predicates = {p.name: p for p in self._initial_predicates}
+        at = predicates["at"]
+        carrying = predicates["carrying"]
+        is_home_base = predicates["ishomebase"]
+        safe = predicates["safe"]
+        satisfied = predicates["satisfied"]
+        wants_paper = predicates["wantspaper"]
+        unpacked = predicates["unpacked"]
+
+        nsrts = {n.name: n for n in self._get_current_nsrts()}
+        deliver = nsrts["deliver"]
+        move = nsrts["move"]
+        pick_up = nsrts["pick-up"]
+
+        paper_var, loc_var = deliver.parameters
+        from_var, to_var = move.parameters
+        paperPick_var, homeBase_var = pick_up.parameters
+        rules = [
+            LDLRule("Pick-up",
+                    parameters=[paperPick_var, homeBase_var, to_var],
```

Comment:
I think you can get away without the `to_var` here and without the goal precondition. In general, it's a good idea to try to minimize the number of parameters in these rules, because that will then reduce the number of ground operator that we'll consider in each state.